.circleci/config.yml

```diff
@@ -23,6 +23,55 @@ executors:
 
 # Custom commands definition
 commands:
+  # Based on https://circleci.com/developer/orbs/orb/kimh/run-with-retry
+  retry_until_success:
```

Comment:
I think the goal of this routine is to retry the entire command so it works for both karma and integration suites. But I was wondering if we can collect only the failed tests and rerun them instead of rerunning the entire command.

Reply:
We could use like a Junit style reporter https://www.npmjs.com/package/karma-junit-reporter and gather the specs that have a failure and rerun those.

Reply:
Some test runners have a built-in retry mechanism, e.g. [Cypress `retries`](https://docs.cypress.io/guides/guides/test-retries#What-you-ll-learn) or [TestCafe quarantine mode](https://testcafe.io/documentation/402830/guides/basic-guides/run-tests#quarantine-mode). Unfortunately I couldn't find anything similar for Karma – if it existed, I would definitely prefer to use it. It does have a [`retryLimit`](http://karma-runner.github.io/6.3/config/configuration-file.html#retrylimit), but this appears to only apply to browser crashes.